app/controllers/voucher_adjustments_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class VoucherAdjustmentsController < ::BaseController
+  include CablecarResponses
+
+  def destroy
+    @order = current_order
+
+    # TODO: do we need to check the user can delete voucher_adjustment
+    @order.voucher_adjustments.find_by(id: params[:id])&.destroy
```

Comment:
That's great. Vouchers are applied by customers and can be removed by them. The current_order always belongs to the logged in user and we are only looking at voucher adjustments here. And since there are no negative vouchers, there's no way to gain more value by deleting a voucher. I think that we are good here.